Vault: ufw for vault

lpaulmp opened the pull request:
Issue #160 

#### Requirements completed:
- Create an ufw profile for vault
- Allow port 8200 for vault server

#### Local testing
```bash
root@ubuntu-xenial:~# salt-call --local state.sls vault.ufw
```
```
local:
----------
          ID: vault-ufw-app-config
    Function: file.managed
        Name: /etc/ufw/applications.d/vault.ufw
      Result: True
     Comment: File /etc/ufw/applications.d/vault.ufw is in the correct state
     Started: 15:15:09.582888
    Duration: 67.972 ms
     Changes:   
----------
          ID: vault-ufw-app-config
    Function: cmd.run
        Name: ufw allow vault
      Result: True
     Comment: Command "ufw allow vault" run
     Started: 15:15:09.652595
    Duration: 111.158 ms
     Changes:   
              ----------
              pid:
                  7574
              retcode:
                  0
              stderr:
              stdout:
                  Rules updated
                  Rules updated (v6)

Summary for local
------------
Succeeded: 2 (changed=1)
Failed:    0
------------
Total states run:     2
Total run time: 179.130 ms
```

ketzacoatl commented on vault/ufw.sls:
```diff
@@ -0,0 +1,13 @@
+
+# description for the service unit and ufw configuration files
+{%- set desc = salt['pillar.get']('vault:systemd_desc', "Hashiscorp Vault") %}
+{%- set server_desc = desc ~ ' Server' %}
```
With vault, we only run the server, there is no client agent, just a cli client. I think it would be ok to include `Server` in the previous line, eg: `{%- set desc = salt['pillar.get']('vault:systemd_desc', "Hashiscorp Vault Server") %}`.

ketzacoatl commented:
@lpaulmp can you post some output from `ufw status ...` that includes 8200 / the entry for vault?

lpaulmp commented:
@ketzacoatl I addressed the comment that you left. 
this is the output of ufw status
```bash
root@ubuntu-xenial:~# ufw status
Status: active

To                         Action      From
--                         ------      ----
nomad-server               ALLOW       Anywhere
nomad-client               ALLOW       Anywhere
vault                      ALLOW       Anywhere
nomad-server (v6)          ALLOW       Anywhere (v6)
nomad-client (v6)          ALLOW       Anywhere (v6)
vault (v6)                 ALLOW       Anywhere (v6)

root@ubuntu-xenial:~# ufw app info vault
Profile: vault
Title: vault
Description: Hashiscorp Vault Server

Port:
  8200
```
